### PR TITLE
Define minimal types involved in tracing

### PR DESCRIPTION
**Pull request by fvictorio**

### CODE REVIEW

File: packages/hardhat-core/src/internal/hardhat-network/provider/vm/vm-adapter.ts [hunk 1]
```diff
@@ -1,17 +1,38 @@
...
+    executionGasUsed: bigint;
+  };
+}
+
+export interface TracingCallbacks {
+  beforeMessage: (message: TracingMessage, next: any) => Promise<void>;
+  step: (step: TracingStep, next: any) => Promise<void>;
+  afterMessage: (result: TracingMessageResult, next: any) => Promise<void>;
+}
+
```
Comment (fvictorio) [inline packages/hardhat-core/src/internal/hardhat-network/provider/vm/vm-adapter.ts:35]: @Wodann this is the relevant part of this change. These callbacks should be called with those values.

We need to figure out if the `next` callbacks will be called from rethnet or from the adapter, but that's a minor detail.

Comment (Wodann) [reply packages/hardhat-core/src/internal/hardhat-network/provider/vm/vm-adapter.ts:35]: Would you be opposed to defining these in `rethnet_evm_nqpi`? Then they are the same between Rethnet and Hardhat automatically

Comment (fvictorio) [reply packages/hardhat-core/src/internal/hardhat-network/provider/vm/vm-adapter.ts:35]: No problem at all!

Comment (fvictorio) [reply packages/hardhat-core/src/internal/hardhat-network/provider/vm/vm-adapter.ts:20]: Done :slightly_smiling_face:

**Review by Wodann — APPROVED** [review]

**Pull request merged**